sqlglot/optimizer/scope.py

```diff
@@ -746,10 +747,16 @@ def _traverse_udtfs(scope):
     scope.sources.update(sources)
 
 
-def _traverse_create(scope):
+def _traverse_ddl(scope):
     yield from _traverse_ctes(scope)
-    yield from _traverse_subqueries(scope)
-    yield from _traverse_tables(Scope(scope.expression.expression))
+
+    query_scope = scope.branch(
+        scope.expression.expression, scope_type=ScopeType.DERIVED_TABLE, chain_sources=scope.sources
+    )
```

Comment:
Picked `DERIVED_TABLE` here because `ROOT` is already taken for the DDL statement itself. Might not be the cleanest approach but I think it's fine for now?